File: android/app/src/main/java/com/airbridge/app/core/interfaces/IDeviceRegistry.kt
package com.airbridge.app.core.interfaces

import com.airbridge.app.core.models.DeviceInfo
import kotlinx.coroutines.flow.Flow

/**
 * Manages the local registry of known and paired devices.
 * Exposes [devicesFlow] as a reactive stream for UI observation.
 * Implementations must be thread-safe.
 */
interface IDeviceRegistry {
    /** Reactive stream of the current device list — emits on every change. */
    val devicesFlow: Flow<List<DeviceInfo>>

    fun getAllDevices(): List<DeviceInfo>
    fun getPairedDevices(): List<DeviceInfo>
    fun addOrUpdate(device: DeviceInfo)
    fun remove(deviceId: String)
    fun isPaired(deviceId: String): Boolean
}

File: windows/AirBridge.Core/Interfaces/IDeviceRegistry.cs
using AirBridge.Core.Models;

namespace AirBridge.Core.Interfaces;

/// <summary>
/// Manages the local registry of known and paired devices.
/// Implementations must be thread-safe.
/// </summary>
public interface IDeviceRegistry
{
    /// <summary>Returns all currently known devices (discovered or paired).</summary>
    IReadOnlyList<DeviceInfo> GetAllDevices();

    /// <summary>Returns only devices that have been successfully paired.</summary>
    IReadOnlyList<DeviceInfo> GetPairedDevices();

    /// <summary>Adds or updates a device in the registry.</summary>
    void AddOrUpdate(DeviceInfo device);

    /// <summary>Removes a device by its ID. Does nothing if not found.</summary>
    void Remove(string deviceId);

    /// <summary>Returns true if a device with the given ID exists and is paired.</summary>
    bool IsPaired(string deviceId);

    /// <summary>Raised when the registry contents change.</summary>
    event EventHandler<DeviceInfo> DeviceChanged;
}
